### PR TITLE
Migrate Image Class Component to Functional Component and hooks

### PR DESCRIPTION
**Pull request by arpitBhalla**
**What kind of change does this PR introduce?**
Convert Class based `Image` Component to `Functional Component`.

**Did you add tests for your changes?**
Yes, I have updated the tests and migrated them to RN-testing Library.

**If relevant, did you update the documentation?**
No changes are required.

**Does this PR introduce a breaking change?**
No, this would not break anything.

**Other information**
GSoC Task of @arpitBhalla.
- [x] Self Review for any kind of error in code.

#### I have tested the PR manually in demo app / Snack
- [x] Android
- [x] iOS
- [x] Web ( Might be solved by using `width` and `height`  )

**This PR only migrate the component to FC**

### CODE REVIEW

File: src/Image/index.tsx [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+
+const ThemedImage = Object.assign(withTheme(Image, 'Image'), {
+  getSize: ImageNative.getSize,
+  getSizeWithHeaders: ImageNative.getSizeWithHeaders,
+  prefetch: ImageNative.prefetch,
+  abortPrefetch: ImageNative.abortPrefetch,
+  queryCache: ImageNative.queryCache,
+  resolveAssetSource: ImageNative.resolveAssetSource,
+});
+
```
Comment (pranshuchittora) [inline src/Image/index.tsx:16]: Can we do something like this 🤔 

```js
{
...withTheme(Image, 'Image'),
  getSize: ImageNative.getSize,
  getSizeWithHeaders: ImageNative.getSizeWithHeaders,
  prefetch: ImageNative.prefetch,
  abortPrefetch: ImageNative.abortPrefetch,
  queryCache: ImageNative.queryCache,
  resolveAssetSource: ImageNative.resolveAssetSource,
}
```

Comment (arpitBhalla) [reply src/Image/index.tsx]: `JSX element type 'Image' does not have any construct or call signatures.`

**Review by pranshuchittora — CHANGES_REQUESTED** [review]
### Image
**Getting error on `PlaceholderContent` prop in the playground.**

Peek 2021-08-06 01-12

**Not able to disable `transition`. It is always enabled.**

https://user-images.githubusercontent.com/32242596/128412075-639fdf65-befe-460e-b78d-4ff697a5ee06.mp4

Pls, check the following issues.

File: src/Image/__tests__/Image.test.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+describe('Image Component', () => {
+  const FAKE_URI = 'https://i.imgur.com/0y8Ftya.jpg';
+  it('should render', () => {
+    const { toJSON } = renderWithTheme(
+      <Image testID="customTestId" source={{ uri: FAKE_URI }} />
+    );
+    expect(toJSON()).toMatchSnapshot();
+  });
+  it('should render the appropriate testId when one is passed.', () => {
```
Comment (pranshuchittora) [inline src/Image/__tests__/Image.test.tsx:16]: Newline after each test (after each `it`)

File: src/Image/__tests__/Image.test.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        },
+      },
+    };
+    const { queryByTestId } = renderWithTheme(
+      <Image source={{ uri: FAKE_URI }} />,
+      textTheme
+    );
+    const placeholder = queryByTestId('RNE__Image__placeholder');
+    expect(placeholder.props.style.backgroundColor).toBe('red');
+  });
```
Comment (pranshuchittora) [inline src/Image/__tests__/Image.test.tsx:44]: Pls add a test for `PlaceholderContent`

File: src/Image/Image.tsx [hunk 4]
```diff
@@ -123,7 +114,7 @@ export class Image extends React.Component<
             {PlaceholderContent}
```
Comment (pranshuchittora) [inline src/Image/Image.tsx:114]: Shouldn't work as a child of `View`

**Comment by arpitBhalla** [comment]
Hello World

File: src/Image/Image.tsx (hunk 4 above)
Comment (arpitBhalla) [reply src/Image/Image.tsx]: `PlaceholderContent` is of type `ReactNode` so user have to use `Text` Component for text, not directly

Comment (arpitBhalla) [reply src/Image/Image.tsx]: PlaceholderContent type is `ReactElement` so user can do `PlaceholderContent={<Text>Hey</Text>}`

File: src/Avatar/__tests__/__snapshots__/Avatar.test.tsx.snap [hunk 5]
```diff
@@ -166,7 +166,6 @@ exports[`Avatar Component should render without issues 1`] = `
           },
         }
       }
-      transition={true}
```
Comment (pranshuchittora) [inline src/Avatar/__tests__/__snapshots__/Avatar.test.tsx.snap:169]: Why there is a change in the behaviour of the Avatar?
I.e. why default transition is turned off?

**Review by pranshuchittora — CHANGES_REQUESTED** [review]
- [ ] Not able to apply transition on avatar.
- [ ] Pls fix the netlify config

File: src/Avatar/__tests__/__snapshots__/Avatar.test.tsx.snap (hunk 5 above)
Comment (arpitBhalla) [reply src/Avatar/__tests__/__snapshots__/Avatar.test.tsx.snap:169]: React Native Image do no have `transition` prop.

**Comment by arpitBhalla** [comment]
> Not able to apply transition on avatar.

Its working sir.

> Pls fix the netlify config 

This PR has nothing to do with netlify, A fixed have been added in `Pressable` PR

**Comment by arpitBhalla** [comment]
> Not able to apply transition on avatar.

`transition` is working in Avatar
> Pls fix the netlify config

I have made changes in `Pressable` PR.

**Pull request merged**